Source/WebKit/WebProcess/WebPage/WebFrame.cpp

```diff
@@ -1321,6 +1321,12 @@ inline DocumentLoader* WebFrame::policySourceDocumentLoader() const
     return policySourceDocumentLoader.get();
 }
 
+void WebFrame::dispatchBeforeUnloadEventForCrossProcessNavigation(CompletionHandler<void(bool)>&& completionHandler)
+{
+    RefPtr coreFrame = coreLocalFrame();
+    completionHandler(!coreFrame || coreFrame->protectedLoader()->shouldClose());
```

Comment:
I think we need to rename shouldClose()

Reply:
Why? What name do you suggest to use instead?